### ISSUE TITLE
lx-brand: user-script's not being run

### ISSUE THREAD

**Issue by sjorge** [joyent/smartos-live#451]
Doesn't look like the user-script's are being run, another job for lxinit? (move from, illumos-joyent #56)

**Comment by chorrell** [comment]
Hi,

Does this happen for all lx-brand images? Can you give me an example script that fails?

**Comment by sjorge** [comment]
I only tried the latest ubuntu one
```
  "customer_metadata": {
    "user-script": "/native/usr/sbin/zfs set mountpoint=/var/lib/plexmediaserver zones/`zonename`/data; /native/usr/sbin/zfs mount -a ; /srv/.bootstrap_zone"
  },
```

This works fine for my other zones (non lx) but nothimg seems to happen for the lx-branded zone. dataset is not mounted, mountpoint not changed and /srv/.bootstrap_zone not executed.

Running that manually after the zone has been booted works, I know normall mdata:execute takes care of running the user script. I don't think there is an equal for lx-branded zones yet.

**Comment by chorrell** [comment]
HI,

Actually, there is.

In /lib/smartdc for lx-brand images there are two relevant scripts called via rc.local (/etc/rc.local on the Ubuntu image):

/lib/smartdc/mdata-fetch
/lib/smartdc/mdata-execute

The mdata-fetch script fetches available meta-data (user-data, user-script, operator-script, oot_authorized_keys). The mdata-execute script executes operator-script and user-script. Logging goes to syslog (/var/log/syslog on Ubuntu), although it's not very verbose.

**Comment by sjorge** [comment]
Let me try again later and see what the logs say. It may just be the whole datasets not mounthing thing that is messing it up then

**Comment by sjorge** [comment]
So because the dataset did not get automatically mounted the script never ran in my initial tests.
I added the /native/usr/sbin/zfs mount -a but did not rerun, on a fresh zone it actually worked.
